pulp_ansible/app/tasks/collections.py

```diff
@@ -81,6 +82,32 @@ def sync(remote_pk, repository_pk, mirror):
     d_version = AnsibleDeclarativeVersion(first_stage, repository, mirror=mirror)
     d_version.create()
 
+    repo_version = repository.latest_version()
+    to_deprecate = []
+    to_deprecate_query = Q()
+    to_undeprecate_query = Q()
+    for collection, deprecated in first_stage.deprecations.items():
+        namespace, name = collection.split(".")
+        if deprecated:
+            to_deprecate_query |= Q(namespace=namespace, name=name)
+        else:
+            to_undeprecate_query |= Q(namespace=namespace, name=name)
+
+    if to_deprecate_query:
+        collections_deprecate_true_qs = Collection.objects.filter(to_deprecate_query)
+        for collection in collections_deprecate_true_qs:
+            to_deprecate.append(
+                AnsibleCollectionDeprecated(repository_version=repo_version, collection=collection)
+            )
+
+        AnsibleCollectionDeprecated.objects.bulk_create(to_deprecate, ignore_conflicts=True)
+
+    if to_undeprecate_query:
+        collections_deprecate_false_qs = Collection.objects.filter(to_undeprecate_query)
```

Comment:
Instead of constructing `to_undeprecate_query` using `Q()` objects, use the `Collection.objects.exclude(to_deprecate_query).